### PR TITLE
feat: upgrade to winterfell 0.5

### PR DESCRIPTION
**Pull request by vlopes11**

### CODE REVIEW

File: src/hash/rpo/mod.rs [hunk 1]
```diff
@@ -91,6 +91,11 @@ const INV_ALPHA: u64 = 10540996611094048183;
 pub struct Rpo256();
 
 impl Hasher for Rpo256 {
+    /// Rpo256 collision resistance is the same as the security level, that is 128-bits. However,
+    /// out setup of the capacity registers might drop it to 126, so here we use a pessimistic
+    /// approach and pick the lowest possible value.
+    const COLLISION_RESISTANCE: u32 = 126;
```
Comment (bobbinth) [inline src/hash/rpo/mod.rs:97]: nit: typo "out" should probably be "our".

Also, I'd probably still set it to 128 and leave a `TODO` to address later as this affects other things (e.g., how we define proof security level).

File: src/merkle/path_set.rs [hunk 2]
```diff
@@ -16,15 +16,30 @@ impl MerklePathSet {
...
+    pub fn with_paths<I>(self, paths: I) -> Result<Self, MerkleError>
+    where
+        I: IntoIterator<Item = (u64, Word, MerklePath)>,
+    {
+        paths
+            .into_iter()
+            .try_fold(self, |mut set, (index, value, path)| {
+                set.add_path(index, value, path)?;
+                Ok(set)
+            })
```
Comment (bobbinth) [inline src/merkle/path_set.rs:42]: @Al-Kindi-0 - this address the comment I left [here](https://github.com/0xPolygonMiden/miden-vm/pull/705#discussion_r1111325948). The interface is slightly different - but it should accomplish the same thing.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left one small inline comment.

Let's wait until Winterfell v0.5 is published and then we can update/merge.

**Review by Al-Kindi-0 — APPROVED** [review]
Looks great, thank you!

File: src/merkle/path_set.rs (hunk 2 above)
Comment (Al-Kindi-0) [reply src/merkle/path_set.rs:42]: Will make the switch once it is merged.

**Review by bobbinth — APPROVED** [review]
All looks good! Thank you!

**Pull request merged**